pyuoi/linear_model/base.py

```diff
@@ -0,0 +1,505 @@
+import abc as _abc
+import six as _six
+import numpy as np
+
+from tqdm import trange
+
+from sklearn.linear_model.base import (
+    LinearModel, _preprocess_data, SparseCoefMixin)
+from sklearn.metrics import r2_score
+from sklearn.model_selection import train_test_split
+from sklearn.utils import check_X_y
+
+from PyUoI import utils
+
+
+class AbstractUoILinearModel(_six.with_metaclass(_abc.ABCMeta, LinearModel, SparseCoefMixin))
+    """An abstract base class for UoI linear model classes
+
+    See Bouchard et al., NIPS, 2017, for more details on UoI-Lasso and the
+    Union of Intersections framework.
+
+    Parameters
+    ----------
+    n_lambdas : int, default 48
```

Comment:
Alternatively, you could take out ```n_lambdas``` and then use any selection method.

Reply:
I think I was working on this in parallel to your review. I have since pulled out Lasso-specific stuff from this ABC, and made a new ABC for regressors. In the end, nothing inside of base.py will know about alpha/lambda/C/etc--regularization parameter and values will be abstracted out, and served up by the concrete class via `get_reg_params`